pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -244,6 +254,24 @@ func (lb *ListenerBuilder) buildWaypointInboundConnectTerminate() *listener.List
 	return lb.buildConnectTerminateListener(routes)
 }
 
+func (lb *ListenerBuilder) findServiceWaypoint(svc *model.Service) host.Name {
+	if lb.node.Type != model.Router && !isEastWestGateway(lb.node) {
+		return ""
+	}
+	if !svc.HasAddressOrAssigned(lb.node.Metadata.ClusterID) {
```

Comment:
nit: do we really need to check that there's a clusterID if we know this is an e/w gateway?

Reply:
Right, this is not needed either - I just copied it from the endpoint_builder.go and didn't drop. Fixed now.